docs/advisors/checks/mongodb-swap-allocation.md

```diff
@@ -0,0 +1,35 @@
+# MongoDB Swap Allocation
+
+## Description
+This check returns a warning if there is no swap memory allocated to your instance.
+
+## Resolution
+
+MongoDB performs best where swapping can be avoided or kept to a minimum since retrieving data from swap on disk will always be slower than accessing data in RAM. However, if the system hosting MongoDB runs out of RAM, swapping can prevent the Linux OOM Killer from terminating the mongod process.
+
+Generally, you should choose one of the following swap strategies:
+
+1. Assign swap space on your system, and configure the kernel to only permit swapping under high memory load, or
+2. Do not assign swap space on your system, and configure the kernel to disable swapping entirely.
+
+If your MongoDB instance is hosted on a system that also runs other software, you should choose the first swap strategy to prevent possible negative impacts to those applications. Do not disable swap in this case. It is highly recommended that you run MongoDB on its own dedicated system whenever possible.
```

Comment:
```suggestion
If your MongoDB instance is hosted on a system that runs other software, you should choose the first swap strategy to prevent possible negative impacts on those applications. Do not disable swap in this case. Percona highly recommends that you run MongoDB on its own dedicated system whenever possible.
```